### PR TITLE
fix(plugin-meetings): handle meetingInfo response without permission token

### PR DESCRIPTION
**Pull request by sreenara**
# COMPLETES # [SPARK-516464](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-516464)

## This pull request addresses

Instant Connect is unable to show details in the app regarding these 2 topics:
* Report a problem and
* MeetingInfo (that displays information to join from video devices)

This is occurring because the meetingInfo object within meeting does not get populated when using SDK 3.0. While using SDK 2.35.0, Instant Connect does not observe the issues stated above.

Instant connect uses the Conversation URL of a space to create an instant meeting and WbxAppAPI service returns the `meetingInfo` response without a permissions token.

This issue is reproducible in the Kitchen sink page of the SDK as well when trying to create a `meeting` object with the conversation URL.

```
logger.js:396 wx-js-sdk Meetings:index#createMeeting --> Info Unable to fetch meeting info for https://conversation-intb.ciscospark.com/conversation/api/v1/conversations/3755da40-fc84-11ee-b4a6-198bb1d302bd.
logger.js:396 wx-js-sdk Meetings:index#createMeeting --> Info assuming this destination is a 1:1 or wireless share
```

## by making the following changes

The `permissionToken` is an optional parameter. Therefore, `setPermissionTokenPayload()` and `setSelfUserPolicies()` functions should also be called only when permissionToken is available. 

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

Get a guest token and conversation URL. Initialize the webex object and create a meeting with the conversation URL. Check if `meeting.meetingInfo` object is populated.

Collected logs and HAR files for both working and fixed scenarios.

[meetingInfo-missing-error.har.txt](https://github.com/webex/webex-js-sdk/files/15007402/meetingInfo-missing-error.har.txt)
[meetingInfo-missing-error-console.log](https://github.com/webex/webex-js-sdk/files/15007403/meetingInfo-missing-error-console.log)
[fix-working-console-permissiontoken.har.txt](https://github.com/webex/webex-js-sdk/files/15007404/fix-working-console-permissiontoken.har.txt)
[fix-working-console-permissiontoken.log](https://github.com/webex/webex-js-sdk/files/15007407/fix-working-console-permissiontoken.log)

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by mkesavan13 — APPROVED** [review]
Looks good to me. Thanks, Sreekanth

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 1]
```diff
@@ -3459,8 +3459,10 @@ export default class Meeting extends StatelessWebexPlugin {
       this.owner =
         locusMeetingObject?.info.owner || meetingInfo?.owner || meetingInfo?.hostId || this.owner;
       this.permissionToken = meetingInfo?.permissionToken;
-      this.setPermissionTokenPayload(meetingInfo?.permissionToken);
-      this.setSelfUserPolicies();
+      if (this.permissionToken) {
+        this.setPermissionTokenPayload(meetingInfo?.permissionToken);
+        this.setSelfUserPolicies();
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:3464]: any idea what does the setSelfUserPolicies do , any relation ship with permissionToken ?

Comment (sreenara) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:3464]: @arun3528 setPermissionTokenPayload decodes the jwt and then user policies are stored from the permission token payload.

Comment (marcin-bazyl) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:3464]: setSelfUserPolicies() uses optional chaining, so it should work even if permissionTokenPayload is undefined:
image

**Review by marcin-bazyl — APPROVED** [review]

**Pull request merged**